packages/webpack/src/buildtime/modulesData.js

```diff
@@ -0,0 +1,242 @@
+ *   knownPaths: { path: string; moduleId: string | number }[]
+ *   unenforceableModuleIds: (string | number)[]
+ *   externals: Record<string | number, string>
+ * }}
+ */
+exports.analyzeModules = ({
+  mainCompilationWarnings,
+  allIdentifiedModules,
+}) => {
+  /**
+   * Array of objects representing the paths and module ids found in the
+   * generation process.
+   *
+   * @type {{ path: string; moduleId: string | number }[]}
+   */
+  const knownPaths = []
+  /**
+   * Array of module ids that are unenforceable by policy.
+   *
+   * @type {(string | number)[]}
+   */
+  const unenforceableModuleIds = []
+  /**
+   * Array of module ids that are context modules and need to be double-wrapped.
+   *
+   * @type {{ moduleId: string | number; context: string }[]}
+   */
+  const contextModules = []
+
+  /**
+   * An array of modules deemed fit for inspecting for policy
+   *
+   * @type {import('./policyGenerator.js').InspectableWebpackModule[]}
+   */
+  const inspectable = []
+
+  /**
+   * A record of module ids that are externals and need to be enforced as
+   * builtins.
+   *
+   * @type {Record<string | number, string>}
+   */
+  const externals = {}
+
+  /**
+   * @param {IdentifiedModule} arg
+   * @returns
+   */
+  const processModule = ({ module, moduleId }) => {
+    const moduleClass = Object.getPrototypeOf(module).constructor.name
+
+    if (moduleId === null) {
+      diag.rawDebug(
+        2,
+        `LavaMoatPlugin: module ${module.identifier()} has no moduleId, cannot cover it with policy.`
+      )
+      diag.rawDebug(4, { module })
+      return
+    }
+
+    // ==================================================
+    // Fixing bad modules
+
+    // Fixes the issue with assets being emitted to dist without a loader
+    // TODO: refactor to move random hardening of the build somewhere it's easier to track.
+    if (
+      isAmbientAsset(module) &&
+      module.resource.includes('node_modules') // FIXME: would be better to use canonicalName lookup and match with root
+    ) {
+      // add a warning about removing the asset
+      mainCompilationWarnings.push(
+        new WebpackError(
+          `LavaMoatPlugin: the following resource was being silently emitted to the dist directory and LavaMoat has prevented it: '${module.resource}'. If you want to add this resource, explicitly define a file-loader for it in your webpack configuration.`
+        )
+      )
+
+      // We can't use `chunkGraph.disconnectChunkAndModule` here
+      // because the require statement remains and errors out
+
+      if (module.generatorOptions) {
+        // generatorOptions was not present in testing, but types indicate it might be there
+        module.generatorOptions.emit = false
+      }
+      if (module.generator) {
+        module.generator = Object.create(module.generator, {
+          emit: {
+            value: false,
+            writable: false,
+            configurable: false,
+            enumerable: true,
+          },
```

Comment:
```suggestion
          emit: {
            value: false,
            enumerable: true,
          },
```

`writable` and `configurable` always default to `false`, fwiw